File: sample/app/src/main/java/com/example/activitystarter/StudentParcelableActivity.java
package com.example.activitystarter;

import android.os.Bundle;
import android.widget.TextView;

import activitystarter.Arg;
import activitystarter.MakeActivityStarter;

@MakeActivityStarter
public class StudentParcelableActivity extends BaseActivity {

    @Arg StudentParcelable student;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_data);
        TextView nameView = (TextView) findViewById(R.id.name_view);
        TextView idView = (TextView) findViewById(R.id.id_view);
        TextView gradeView = (TextView) findViewById(R.id.grade_view);

        nameView.setText("Name: "+student.getName());
        idView.setText("Id: "+student.getId());
        gradeView.setText("Grade: "+student.getGrade());
    }
}

File: sample/publicLibKotlin/src/main/java/com/marcinmoskala/kotlinapp/StudentParcelableActivity.kt
package com.marcinmoskala.kotlinapp

import android.os.Bundle
import android.widget.TextView

import activitystarter.Arg
import activitystarter.MakeActivityStarter
import kotlinx.android.synthetic.main.activity_data.*

@MakeActivityStarter
class StudentParcelableActivity : BaseActivity() {

    @Arg var student: StudentParcelable? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_data)

        nameView.text = "Name: " + student!!.name
        idView.text = "Id: " + student!!.id
        gradeView.text = "Grade: " + student!!.grade
    }
}
